### PR TITLE
Make `/coin check` default to the caller

### PR DESCRIPTION
**Pull request by hydrobeam**
# Related Issues

N/A

# Summary of Changes

In comparison to `.coin`, doing `/coin check` and having to look for the target user was pretty time consuming when somebody just wanted to check their own balance. This PR makes the `user` argument optional and defaults to checking the balance of the user who called the command if the arg is not provided. 

- Balance not present + coin check optional
image

- .coin default behaviour checks out + message changes depending on how user is called
image

# Steps to Reproduce

`/coin check` and pass in no arguments.
OR
`.coin check` and pass in no arguments.

## Extra commentary

Not sure if `messageFromUser.member?.user` is the "right" way to get the user, since for some reason accessing `member` can return null from a `Message`?🤷

### CODE REVIEW

File: src/commandDetails/coin/check.ts
Comment (mcpenguin) [inline src/commandDetails/coin/check.ts:20]: Does `messageFromUser.user` work?

**Comment by mcpenguin** [comment]
Also please leave screenshots :))

File: src/commandDetails/coin/check.ts
Comment (hydrobeam) [reply src/commandDetails/coin/check.ts:20]: Nope, `user` is not a valid field for a `Message`. `author` exists for message, but I don't really know in what situation accessing member could be null for a message 🤷 so it's probably not a concern

**Comment by hydrobeam** [comment]
Decided to remove the `balance` subcommand since having `/coin check` default to self makes it redundant. To maintain appearances, I added a check to see if the caller is running the method without arguments and changes the initial message to `You have` accordingly.

Also reworked `getUserIDFromMessage` to be more general

File: src/commandDetails/coin/check.ts
Comment (mcpenguin) [inline src/commandDetails/coin/check.ts:43]: If you're removing balance, make `aliases` also have `balance, bal and b`, since people use these commands regularly.

**Review by mcpenguin — CHANGES_REQUESTED** [review]
small changes

File: src/commandDetails/coin/check.ts
Comment (hydrobeam) [reply src/commandDetails/coin/check.ts:43]: Done

**Review by mcpenguin — APPROVED** [review]
LGTM! Feel free to merge after you fix the linter.

**Comment by hydrobeam** [comment]
> LGTM! Feel free to merge after you fix the linter.

Fixed the linter, but don't have write access :see_no_evil:

**Pull request merged**
